### PR TITLE
BRUCE-6694: Add the wait flag to redis and remove a read to the redis before SET and change it with SETNX

### PR DESCRIPTION
**Pull request by Shahid097**
Add a Wait flag to the DBConfig to let the request for connection be available, Remove the read of the key before the Set to DB and using SETNX for adding the key

### CODE REVIEW

**Review by amar-shalgar — APPROVED** [review]

**Review by sriteja-sabbineni — APPROVED** [review]

File: lib-persistence-manager/persistencemgr/redis.go [hunk 1]
```diff
@@ -394,7 +389,7 @@ func (p *ConnPool) Update(table, key string, data interface{}) (string, *errors.
 	}
 	writeConn := writePool.Get()
 	defer writeConn.Close()
-	_, createErr := writeConn.Do("SET", saveID, jsondata)
+	_, createErr := writeConn.Do("SETNX", saveID, jsondata)
```
Comment (amar-shalgar) [inline lib-persistence-manager/persistencemgr/redis.go:392]: should this be SETXX or SETNX, check once

**Review by amar-shalgar — CHANGES_REQUESTED** [review]

File: lib-persistence-manager/persistencemgr/redis.go [hunk 2]
```diff
@@ -253,6 +253,7 @@ func getPool(host, port, password string) (*redis.Pool, error) {
 		MaxActive: config.Data.DBConf.MaxActiveConns,
 		// Dial is an application supplied function for creating and
 		// configuring a connection.
+		Wait: true,
```
Comment (Bharath-KKB) [inline lib-persistence-manager/persistencemgr/redis.go:256]: what does this do

File: lib-persistence-manager/persistencemgr/redis.go [hunk 3]
```diff
@@ -373,12 +374,6 @@ func (p *ConnPool) Create(table, key string, data interface{}) *errors.Error {
 */
 func (p *ConnPool) Update(table, key string, data interface{}) (string, *errors.Error) {
```
Comment (Bharath-KKB) [inline lib-persistence-manager/persistencemgr/redis.go:376]: previously it was informed this code has already gone in. Why is this getting merged now

File: lib-persistence-manager/persistencemgr/redis.go (hunk 1 above)
Comment (Bharath-KKB) [reply lib-persistence-manager/persistencemgr/redis.go:392]: we should use XX for this

File: lib-persistence-manager/persistencemgr/redis.go [hunk 4]
```diff
@@ -373,12 +374,6 @@ func (p *ConnPool) Create(table, key string, data interface{}) *errors.Error {
...
```
Comment (Bharath-KKB) [inline lib-persistence-manager/persistencemgr/redis.go:375]: correct spelling mistakes in the comment

**Pull request closed without merging**